docs/src/modules/components/Link.tsx

```diff
@@ -0,0 +1,98 @@
+/* eslint-disable jsx-a11y/anchor-has-content */
+import * as React from 'react';
+import clsx from 'clsx';
+import { useRouter } from 'next/router';
+import NextLink, { LinkProps as NextLinkProps } from 'next/link';
+import MuiLink, { LinkProps as MuiLinkProps } from '@material-ui/core/Link';
+import { useUserLanguage } from 'docs/src/modules/utils/i18n';
+
+type NextLinkComposedProps = Omit<React.AnchorHTMLAttributes<HTMLAnchorElement>, 'href'> &
+  Omit<NextLinkProps, 'href' | 'as'> & {
+    to: NextLinkProps['href'];
+    linkAs?: NextLinkProps['as'];
+    href?: any;
```

Comment:
```suggestion
    href?: unknown;
```
Should be preferred nowadays for soundness. Where we have unsound types an explicit cast signals intent better.